backend/Dockerfile

```diff
@@ -1,25 +1,29 @@
-# Use node v18 docker image
-FROM node:18
+FROM node:18 AS base
 
-# Create app directory
-WORKDIR /app
+# The builder layer will be used to build the application
+FROM base AS builder
 
-COPY . .
+WORKDIR /app
 
-# Install app dependencies
-# A wildcard is used to ensure both package.json AND package-lock.json are copied
-# where available (npm@5+)
+# The dependencies will be cached unless package.json changes.
 COPY package*.json ./
-
 RUN npm ci
 
-# Build application
+COPY . .
+
+# And the release layer will be used to run the application.
+# Only required files will be copied from the builder layer to the release layer.
 RUN npm run build
 
-# Remove dev-dependencies
-RUN npm ci --only=production && npm cache clean --force 
+FROM base AS release
+
+WORKDIR /app
+
+# If additional files are required, they can be copied from the builder layer to the release layer.
+COPY --from=builder /app/package*.json ./
+COPY --from=builder /app/dist ./dist
+COPY --from=builder /app/node_modules ./node_modules
```

Comment:
Doesn't this also copy all dev dependencies?

Reply:
Yes

Reply:
Doesn't this bloat the image (a bit)?

Reply:
Yeah its not optimal for image size but its faster for build time. Depends on what we are trying to optimize here..